ambry-store/src/test/java/com.github.ambry.store/OffsetTest.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.store;
+
+import com.github.ambry.utils.ByteBufferInputStream;
+import com.github.ambry.utils.Utils;
+import com.github.ambry.utils.UtilsTest;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Random;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+
+/**
+ * Tests for {@link Offset}.
+ */
+public class OffsetTest {
+
+  /**
+   * Tests serialization and deserialization of an {@link Offset} class.
+   * @throws IOException
+   */
+  @Test
+  public void offsetSerDeTest()
+      throws IOException {
+    String name = UtilsTest.getRandomString(10);
+    long offset = Utils.getRandomLong(new Random(), Long.MAX_VALUE);
+    Offset logOffset = new Offset(name, offset);
+    byte[] serialized = logOffset.toBytes();
+    Offset deserializedOffset =
+        Offset.fromBytes(new DataInputStream(new ByteBufferInputStream(ByteBuffer.wrap(serialized))));
+    assertEquals("Original offset 'name' does not match with the deserialized offset", logOffset.getName(),
+        deserializedOffset.getName());
+    assertEquals("Original offset 'offset' does not match with the deserialized offset", logOffset.getName(),
+        deserializedOffset.getName());
+    // equals test
+    assertEquals("Original offset does not match with the deserialized offset", logOffset, deserializedOffset);
```

Comment:
this is the only check that should be done in theory - equality of individual fields (or not) is up to the `Offset.equals()` method.